src/main/java/com/exactpro/th2/fix/client/Main.java

```diff
@@ -196,22 +194,26 @@ public static void run(Settings settings, MessageRouter<MessageGroupBatch> messa
                 try {
                     if (group.getMessagesCount() != 1) {
                         if (LOGGER.isErrorEnabled()) {
-                            LOGGER.error("Message group contains more than 1 message {} ", toJson(group));
+                            LOGGER.error("Message group contains more or less than 1 message {} ", toJson(group));
                         }
                     } else {
                         AnyMessage message = group.getMessagesList().get(0);
                         if (!message.hasRawMessage()) {
                             if (LOGGER.isErrorEnabled()) {
                                 LOGGER.error("Message in the group is not a raw message {} ", toJson(message));
                             }
-                        } else {
-                            String sessionAlias = MessageUtil.getSessionAlias(message);
-                            String strMessage = MessageUtil.rawToString(message);
-                            Session session = Session.lookupSession(sessionIDS.get(sessionAlias));
-
-                            Message fixMessage = MessageUtils.parse(session, strMessage);
-                            session.send(fixMessage);
+                            return;
+                        }
+                        String sessionAlias = MessageUtil.getSessionAlias(message);
+                        if (sessionAlias == null || sessionAlias.isBlank()) {
+                            throw new IllegalArgumentException("No such session alias for message: " + message);
                         }
```

Comment:
Instead we should check `sessionIDs.get(sessionAlias)` for `null`
I also doubt that `sessionAlias` retrieved from gRPC message could ever be `null`